webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/activity/ActivityStreamActivity.vue

```diff
@@ -235,12 +235,12 @@ export default {
     initialized() {
       if (this.initialized && !this.isActivityShared) {
         this.unreadMetadata = this.activity?.metadatas?.unread?.length && this.activity?.metadatas?.unread[0];
-        const isLikeAction = this.activity?.metadatas?.unread[0]?.properties?.actionType === 'Like'
-            || this.activity?.metadatas?.unread[0]?.properties?.actionType === 'LikeComment';
-        const isNewCommentAction = this.activity?.metadatas?.unread[0]?.properties?.actionType === 'ActivityComment'
-            || this.activity?.metadatas?.unread[0]?.properties?.actionType === 'ActivityReplyToComment'
-            || this.activity?.metadatas?.unread[0]?.properties?.actionType === 'EditComment'
-            || this.activity?.metadatas?.unread[0]?.properties?.actionType === 'GamificationActionAnnouncedNotification';
+        const isLikeAction = this.unreadMetadata && this.activity?.metadatas?.unread[0]?.properties?.actionType === 'Like'
+            || this.unreadMetadata && this.activity?.metadatas?.unread[0]?.properties?.actionType === 'LikeComment';
+        const isNewCommentAction = this.unreadMetadata && this.activity?.metadatas?.unread[0]?.properties?.actionType === 'ActivityComment'
+            || this.unreadMetadata && this.activity?.metadatas?.unread[0]?.properties?.actionType === 'ActivityReplyToComment'
+            || this.unreadMetadata && this.activity?.metadatas?.unread[0]?.properties?.actionType === 'EditComment'
+            || this.unreadMetadata && this.activity?.metadatas?.unread[0]?.properties?.actionType === 'GamificationActionAnnouncedNotification';
```

Comment:
Please use extensionRegistry extensibility to determine which behavior to adopt for each activity type instead of embedding business logic of other addons in social which makes the code unmaintainable and not flexible enough for other DAO members (eXo addons)

Reply:
done